erpnext/projects/doctype/timesheet/timesheet.js

```diff
@@ -209,17 +226,23 @@ var calculate_end_time = function(frm, cdt, cdn) {
 
 	let d = moment(child.from_time);
 	if(child.hours) {
-		d.add(child.hours, "hours");
-		frm._setting_hours = true;
-		frappe.model.set_value(cdt, cdn, "to_time",
-			d.format(frappe.defaultDatetimeFormat)).then(() => {
-				frm._setting_hours = false;
-			});
-	}
-
-
-	if((frm.doc.__islocal || frm.doc.__onload.maintain_bill_work_hours_same) && child.hours){
-		frappe.model.set_value(cdt, cdn, "billing_hours", child.hours);
+		var time_diff = (moment(child.to_time).diff(moment(child.from_time),"seconds")) / (60 * 60 * 24);
```

Comment:
business logic should be server side!

https://github.com/frappe/erpnext/wiki/Pull-Request-Checklist#all-business-logic-and-validations-must-be-on-server-side